### PR TITLE
added info on clang environment variables

### PR DESCRIPTION
**Pull request by richardelling**

### CODE REVIEW

**Comment by brendangregg** [comment]
Thanks; I'd move this to the Usage section at the start, as section "9. Clang Environment" after "8. Environment Variables". Plus add the link in the TOC. :)

**Comment by richardelling** [comment]
sure, that makes sense

File: docs/reference_guide.md [hunk 1]
```diff
@@ -390,6 +391,15 @@ Default: 4096
 
 This is the maximum number of keys that can be stored in a map. Increasing the value will consume more memory and increase startup times. There are some cases where you will want to: for example, sampling stack traces, recording timestamps for each page, etc.
 
+## 9. Clang Environment Variables
+
+bpftrace compiles programs with clang. Thus environment variable affecting
```
Comment (ajor) [inline docs/reference_guide.md]: Could you clarify this to say something like "bpftrace parses C header files using libclang". We don't actually use it to compile anything

Comment (richardelling) [reply docs/reference_guide.md]: sure

File: docs/reference_guide.md [hunk 2]
```diff
@@ -390,6 +391,15 @@ Default: 4096
 
 This is the maximum number of keys that can be stored in a map. Increasing the value will consume more memory and increase startup times. There are some cases where you will want to: for example, sampling stack traces, recording timestamps for each page, etc.
 
+## 9. Clang Environment Variables
+
+bpftrace bpftrace parses C header files using libclang, the C interface to Clang.
```
Comment (ajor) [inline docs/reference_guide.md]: There's a double "bpftrace" at the start of the line. Looks good otherwise

Comment (richardelling) [reply docs/reference_guide.md]: D'oh!

**Comment by ajor** [comment]
Thanks!

**Pull request merged**